### PR TITLE
Adding sanity test

### LINKED ISSUE #675: Replace robo tests with sanity activity tests
While porting the application in this repository to a newer maps sdk version. It shows that current robo tests do not cover testing the application fully. Some examples are crashing but this isn't captured by the firebase robo tests. To have more control over the reach of our tests on CI I'm suggesting creating activity sanity tests instead. These tests are generated and verify if an activity can be opened without crashing.

cc @langsmith @zugaldia

### PR DESCRIPTION
**Pull request by langsmith**
Resolves #675 by adding sanity tests to make sure that each example can be opened and that a Mapbox map appears. Will be immensely helpful in testing the demo app to catch errors and crashes as the Mapbox Maps SDK for Android continues to be developed.

The work in this pr is largely based off of the sanity test setup in https://github.com/mapbox/mapbox-gl-native.

cc @tobrun @zugaldia

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -50,13 +50,19 @@ jobs:
...
           command: |
             #!/bin/bash
-            echo "Build seperate test apk:"
+            gcloud firebase test android run --type instrumentation \
             ./gradlew :MapboxAndroidDemo:assembleGpservicesDebug --continue
-            gcloud firebase test android run --type robo --app MapboxAndroidDemo/build/outputs/apk/gpservices/debug/MapboxAndroidDemo-gpservices-debug.apk --device-ids shamu --os-version-ids 22 --locales en --orientations portrait --timeout 5m
+            ./gradlew :MapboxAndroidDemo:assembleAndroidTest --continue
+            --app MapboxAndroidDemo/build/outputs/apk/gpservices/debug/MapboxAndroidDemo-gpservices-debug.apk
+            --test MapboxAndroidDemo/build/outputs/apk/gpservices/debug/MapboxAndroidDemo-gpservices-androidTest.apk
+            --device-ids shamu --os-version-ids 22 --locales en --orientations portrait --timeout 5m
```
[the remark this answers is not in the document]
Comment (langsmith) [reply circle.yml]: Sure, will switch to `sailfish` (Pixel I believe) like we use in `gl-native`

File: node_modules/ejs/Jakefile [hunk 2]
```diff
@@ -0,0 +1,70 @@
+var fs = require('fs');
```
Comment (tobrun) [inline node_modules/ejs/Jakefile]: can we depend on an ejs binary instead of including node modules ourselves?

**Comment by tobrun** [comment]
@langsmith, CI is failing with:

> ERROR: (gcloud.firebase.test.android.run) argument --device: Bad syntax for dict arg: [sailfish]. Please see `gcloud topic escaping` if you would like information on escaping list or dictionary flag values.
Usage: gcloud firebase test android run [ARGSPEC] [optional flags]
  optional flags may be  --app | --app-initial-activity | --app-package |
                         --async | --auto-google-login | --device |
                         --device-ids | --directories-to-pull |
                         --environment-variables | --filter | --help | --limit |
                         --locales | --max-depth | --max-steps | --obb-files |
                         --orientations | --os-version-ids | --page-size |
                         --performance-metrics | --record-video |
                         --results-bucket | --results-dir |
                         --results-history-name | --robo-directives |
                         --sort-by | --test | --test-package |
                         --test-runner-class | --test-targets | --timeout |
                         --type | --use-orchestrator

This means that the firebase command you are trying to execute has malformed syntax for the `--device` flag. Looking into the configurations below the message, it it seems that `--device` needs to be changed to `--device-ids`. This should fix the issue, lmk if you run into other issues.

**Comment by langsmith** [comment]
Thanks @tobrun . Yea, I'd tried the different device syntax but have now switched back to `--device-ids` like you suggested above. No stuck with the cause for previous CI failures...

>Uploading [MapboxAndroidDemo/build/outputs/apk/gpservices/debug/MapboxAndroidDemo-gpservices-debug.apk] to Firebase Test Lab...
Uploading [MapboxAndroidDemo/build/outputs/apk/gpservices/debug/MapboxAndroidDemo-gpservices-androidTest.apk] to Firebase Test Lab...
ERROR: (gcloud.firebase.test.android.run) [MapboxAndroidDemo/build/outputs/apk/gpservices/debug/MapboxAndroidDemo-gpservices-androidTest.apk] not found or not accessible

**Comment by tobrun** [comment]
@langsmith that means that the `./gradlew :MapboxAndroidDemo:assembleAndroidTest --continue` command is generating a different apk as `MapboxAndroidDemo/build/outputs/apk/gpservices/debug/MapboxAndroidDemo-gpservices-androidTest.apk`.

There are two ways to resolve this:
 - either run above gradle command locally and use the correct name/path
 - save the complete build folder as circle ci artefact and validate name/path after

**Comment by tobrun** [comment]
image

**Comment by tobrun** [comment]
Updated path shown in https://github.com/mapbox/mapbox-android-demo/pull/741#issuecomment-405583952 with https://github.com/mapbox/mapbox-android-demo/pull/741/commits/541d854e759dc72ab86d1894c7b08ca95349a4d2

**Comment by tobrun** [comment]
Tests are now failing with:

```
│ Skipped │ sailfish-22-en-portrait │ Incompatible device/OS combination │
```

addressed with https://github.com/mapbox/mapbox-android-demo/pull/741/commits/9e8dd3bdffdb02ebecebd0c60420f31c20e9308a

**Comment by langsmith** [comment]
Nevermind about the comment above. Example location is no longer heavily linked to android docs. See https://github.com/mapbox/android-docs/issues/560#issuecomment-405751168

**Comment by johnwilde** [comment]
@langsmith Not sure if this is where to ask this, but I am trying to write an Espresso test using a similar approach to your sanity tests.  However, it seems that unless the app has cached data the `onMapReady` callback in the idling resource is never hit. I'm curious if you experienced this in your Firebase tests?

**Comment by zugaldia** [comment]
@johnwilde great question - not sure if this is an issue but I think it's worth cutting a ticket on http://github.com/mapbox/mapbox-gl-native for the team to review to make sure there isn't some sort of hidden bug. It'd be great if you could include as much information as possible (like code snippets, your testing infrastructure, specific OS versions you're targeting, testing dependencies, etc). Thanks!

**Pull request merged**